packages/worker-relay/src/sqlite/sqlite-relay.ts

```diff
@@ -132,69 +132,82 @@ export class SqliteRelay extends EventEmitter<RelayHandlerEvents> implements Rel
   #insertEvent(db: Database, ev: NostrEvent) {
     if (this.#seenInserts.has(ev.id)) return false;
 
-    const legacyReplacable = [0, 3, 41];
-    if (legacyReplacable.includes(ev.kind) || (ev.kind >= 10_000 && ev.kind < 20_000)) {
-      const oldEvents = db.selectValues("select id from events where kind = ? and pubkey = ? and created <= ?", [
-        ev.kind,
-        ev.pubkey,
-        ev.created_at,
-      ]) as Array<string>;
+    const legacyReplaceableKinds = [0, 3, 41];
+
+    // Handle legacy and standard replaceable events (kinds 0, 3, 41, 10000-19999)
+    if (legacyReplaceableKinds.includes(ev.kind) || (ev.kind >= 10_000 && ev.kind < 20_000)) {
+      const oldEvents = db.selectValues(
+        `SELECT id FROM events WHERE kind = ? AND pubkey = ?`,
```

Comment:
Missing the `created` check, if we insert RE's out of order or we get older version last we will be inserting that one over the newer event already stored

Reply:
@v0l added for both PRE and RE old events check.